tools/fix_timepoint_date_problems.php

```diff
@@ -154,7 +154,7 @@
     return false;
 }
 // check $candID
-if (!preg_match("/^([0-9]{1,10})$/", $candID)) {
+if (!preg_match("/^([0-9]{1,10})$/", strval($candID))) {
```

Comment:
I think this whole if statement should just be replaced by:

`assert($candID instanceof \LORIS\StudyEntities\Candidate\CandID);`

The CandID object has already validated that it's a valid CandID from the constructor. Asserting that it's a CandID (which is what the error message says it is) should be sufficient to be sure we didn't get here some other type (like int or string) in some code path.

Reply:
changed